owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/RemoteShare.kt

```diff
@@ -104,19 +104,19 @@ class RemoteShare : Parcelable, Serializable {
 
     fun readFromParcel(source: Parcel) {
         id = source.readLong()
-        shareWith = source.readString()
-        path = source.readString()
-        token = source.readString()
-        sharedWithDisplayName = source.readString()
-        sharedWithAdditionalInfo = source.readString()
-        name = source.readString()
-        shareLink = source.readString()
+        shareWith = source.readString().toString()
+        path = source.readString().toString()
+        token = source.readString().toString()
+        sharedWithDisplayName = source.readString().toString()
+        sharedWithAdditionalInfo = source.readString().toString()
+        name = source.readString().toString()
+        shareLink = source.readString().toString()
         fileSource = source.readLong()
         itemSource = source.readLong()
+        shareType = ShareType.NO_SHARED
         try {
-            shareType = ShareType.valueOf(source.readString())
+            shareType = source.readString()?.let { ShareType.valueOf(it) }
         } catch (x: IllegalArgumentException) {
-            shareType = ShareType.NO_SHARED
```

Comment:
Is it guaranteed that this case will not occurs in no circumstances?

Reply:
Now it's the default L116 
This way I can get rid of `else` part of `source.readString()?.` L118